packages/world/src/systems/CraftSystem.sol

```diff
@@ -40,17 +40,18 @@ contract CraftSystem is System {
 
     // Require that the entity has all the ingredients in its inventory
     // And delete the ingredients from the inventory as they are used
-    // uint128 totalInputObjectMass = 0;
-    // uint128 totalInputObjectEnergy = 0;
     for (uint256 i = 0; i < recipeData.inputTypes.length; i++) {
       ObjectTypeId inputObjectTypeId = ObjectTypeId.wrap(recipeData.inputTypes[i]);
-      // totalInputObjectMass += ObjectTypeMetadata._getMass(inputObjectTypeId);
-      // totalInputObjectEnergy += ObjectTypeMetadata._getEnergy(inputObjectTypeId);
       if (inputObjectTypeId.isAny()) {
         InventoryUtils.removeAny(caller, inputObjectTypeId, recipeData.inputAmounts[i]);
       } else {
         InventoryUtils.removeObject(caller, inputObjectTypeId, recipeData.inputAmounts[i]);
       }
+
+      // TODO: add a time cost to burning the coal
```

Comment:
what's the context for this?

Reply:
The goal is to give coal utility, see the messages here: https://discord.com/channels/865335009915961364/1338604056540614656/1359240443589562670

lmk if you have more questions!